src/ompparser.yy

```diff
@@ -3438,28 +3432,60 @@ int yywrap()
 OpenMPDirective* parseOpenMP(const char* _input, void * _exprParse(const char*)) {
     printf("Start parsing...\n");
     OpenMPBaseLang base_lang = Lang_C;
-    lang = Lang_C;
-    exprParse = _exprParse;
-    current_directive = NULL;
-    std::string input_string;
-    const char *input = _input;
-    // Since we can't guarantee the input has been preprocessed, it should be checked here.
-    std::regex fortran_regex ("[!][$][Oo][Mm][Pp]");
-    input_string = std::string(input, 5);
-    if (std::regex_match(input_string, fortran_regex)) {
-        base_lang = Lang_Fortran;
-        lang = Lang_Fortran;
-        input_string = std::string(input);
-        std::transform(input_string.begin(), input_string.end(), input_string.begin(), ::tolower);
-        input = input_string.c_str();
-    };
-    //depend_iterators_definition_class = new std::vector<std::vector<const char*>* >();
-    start_lexer(input);
-    int res = yyparse();
-    //depend_iterators_definition_class.clear();
-    end_lexer();
-    if (current_directive) {
-        current_directive->setBaseLang(base_lang);
-    };
-    return current_directive;
+    if (user_set_lang == Lang_unknown){
```

Comment:
Please try to reuse the common code between two IF branches rather than duplicate the whole block directly.